modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/BaseServerComponent.java

```diff
@@ -143,6 +144,12 @@ public AvailabilityType getAvailability() {
                 previousAvailabilityType = DOWN;
                 availabilityType = DOWN;
                 securityRealmReady = false;
+            } catch (UnauthorizedException e) {
+                previousAvailabilityType = DOWN;
+                availabilityType = DOWN;
+                throw new InvalidPluginConfigurationException("User authenticated via HTTP management interface does"
```

Comment:
Instead of throwing the exception here, you could set a boolean (see `securityRealmReady` above). Then you throw the IPCE later, which avoids to catch/rethrow